### ISSUE TITLE
Possible false positives in package "dependents" lists due to shared package name 

### ISSUE THREAD

**Issue by vgherard** [r-universe-org/help#351]
This is probably most easily explained through the example that made me realise this.

I had noticed that one of my packages, called [r2r](https://vgherard.r-universe.dev/r2r), had a curiously high number of dependents. When looking into the [actual list of dependents](https://r-universe.dev/search/?q=needs%3Ar2r), I noticed that most of them came from the same group of package authors. 

Picking one of them, e.g. [{odbc32}](https://hsonne.r-universe.dev/odbc32), and looking into its actual dependencies from the [DESCRIPTION](https://github.com/hsonne/odbc32/blob/master/DESCRIPTION) file, I finally realized that this package does not actually depend on my CRAN package `r2r`, but rather on a remote package (only on Github), that just happens to have the same name.

I guess this is somewhat a corner case, but you might wanted to know.

Thanks for the hard work with r-universe!

**Comment by jeroen** [comment]
Yes package name collisions are tricky, fortunately rare. 

Honestly I think those other people should rename their `r2r` package because it conflicts with your (cran) package. Which is also dangerous for them, because it is very likely that a user will (accidentally) install the CRAN r2r, for example if you run `update.packages("r2r")` their package will get replaced by your r2r.

However it looks like their r2r is not updated for 5 years so maybe there is little they can do about out.

**Comment by vgherard** [comment]
I totally agree with you, and I assume these cases are so rare that a workaround would not be worth the work required.

Btw, thanks, I didn't realize that `update.packages()` would work this way... that's awful!
